Unable to start the app on pre-lollipop devices

**Summary:** 
On pre-lollipop devices (tested with physical device and emulator) the app never starts. Crashes immediately.

**Steps to reproduce:** 
Just start the app on a pre-lollipop device

**Add System logs:**
```
02-28 12:29:37.486 2502-2502/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: fr.free.nrw.commons.debug, PID: 2502
                                                 java.lang.RuntimeException: Unable to get provider fr.free.nrw.commons.modifications.ModificationsContentProvider: java.lang.ClassNotFoundException: Didn't find class "fr.free.nrw.commons.modifications.ModificationsContentProvider" on path: DexPathList[[zip file "/data/app/fr.free.nrw.commons.debug-1.apk"],nativeLibraryDirectories=[/data/app-lib/fr.free.nrw.commons.debug-1, /system/lib]]
                                                     at android.app.ActivityThread.installProvider(ActivityThread.java:4793)
                                                     at android.app.ActivityThread.installContentProviders(ActivityThread.java:4385)
                                                     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4325)
                                                     at android.app.ActivityThread.access$1500(ActivityThread.java:135)
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                     at android.os.Looper.loop(Looper.java:136)
                                                     at android.app.ActivityThread.main(ActivityThread.java:5017)
                                                     at java.lang.reflect.Method.invokeNative(Native Method)
                                                     at java.lang.reflect.Method.invoke(Method.java:515)
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
                                                     at dalvik.system.NativeStart.main(Native Method)
                                                  Caused by: java.lang.ClassNotFoundException: Didn't find class "fr.free.nrw.commons.modifications.ModificationsContentProvider" on path: DexPathList[[zip file "/data/app/fr.free.nrw.commons.debug-1.apk"],nativeLibraryDirectories=[/data/app-lib/fr.free.nrw.commons.debug-1, /system/lib]]
                                                     at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                     at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
                                                     at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
                                                     at android.app.ActivityThread.installProvider(ActivityThread.java:4778)
                                                     at android.app.ActivityThread.installContentProviders(ActivityThread.java:4385) 
                                                     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4325) 
                                                     at android.app.ActivityThread.access$1500(ActivityThread.java:135) 
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256) 
                                                     at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                     at android.os.Looper.loop(Looper.java:136) 
                                                     at android.app.ActivityThread.main(ActivityThread.java:5017) 
                                                     at java.lang.reflect.Method.invokeNative(Native Method) 
                                                     at java.lang.reflect.Method.invoke(Method.java:515) 
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) 
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) 
                                                     at dalvik.system.NativeStart.main(Native Method) 
```


**Device and Android version:** 
4.4

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -48,10 +48,11 @@ dependencies {
     compile 'com.facebook.fresco:fresco:1.3.0'
     compile 'com.facebook.stetho:stetho:1.5.0'
 
-    compile 'com.android.support:multidex:1.0.2'
+    compile 'com.android.support:multidex:1.0.3'
 
     testCompile 'junit:junit:4.12'
     testCompile 'org.robolectric:robolectric:3.7.1'
+    testCompile "org.robolectric:multidex:3.4.2"
 
     testCompile 'com.squareup.okhttp3:mockwebserver:3.8.1'
     androidTestCompile 'com.squareup.okhttp3:mockwebserver:3.8.1'

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -48,6 +48,8 @@ dependencies {
     compile 'com.facebook.fresco:fresco:1.3.0'
     compile 'com.facebook.stetho:stetho:1.5.0'
 
+    compile 'com.android.support:multidex:1.0.2'
+
     testCompile 'junit:junit:4.12'
     testCompile 'org.robolectric:robolectric:3.7.1'
 

--- a/app/src/main/java/fr/free/nrw/commons/CommonsApplication.java
+++ b/app/src/main/java/fr/free/nrw/commons/CommonsApplication.java
@@ -4,6 +4,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.database.sqlite.SQLiteDatabase;
+import android.support.multidex.MultiDexApplication;
 
 import com.facebook.drawee.backends.pipeline.Fresco;
 import com.facebook.stetho.Stetho;
@@ -40,7 +41,7 @@ import timber.log.Timber;
         resDialogCommentPrompt = R.string.crash_dialog_comment_prompt,
         resDialogOkToast = R.string.crash_dialog_ok_toast
 )
-public class CommonsApplication extends Application {
+public class CommonsApplication extends MultiDexApplication {
 
     @Inject SessionManager sessionManager;
     @Inject DBOpenHelper dbOpenHelper;

